src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -0,0 +1,440 @@
+      columnResizing: {
+        ...newState.columnResizing,
+        startX: clientX,
+        columnWidth,
+        nextColumnWidth,
+        headerIdWidths,
+        nextHeaderIdWidths,
+        isResizingColumn: columnId,
+      },
+    };
+  }
+
+  if (action.type === actions.columnResizing) {
+    const { clientX } = action;
+    const {
+      startX = 0,
+      columnWidth = 1,
+      nextColumnWidth = 1,
+      headerIdWidths = [],
+      nextHeaderIdWidths = [],
+    } = newState.columnResizing;
+
+    const deltaX = clientX - startX;
+
+    const newColumnWidths = getColumnWidths(
+      headerIdWidths,
+      deltaX / columnWidth,
+    );
+    const newNextColumnWidths = getColumnWidths(
+      nextHeaderIdWidths,
+      -deltaX / nextColumnWidth,
+    );
+
+    if (
+      !isNewColumnWidthsValid(newColumnWidths, instance?.flatHeaders) ||
+      !isNewColumnWidthsValid(newNextColumnWidths, instance?.flatHeaders)
+    ) {
+      return newState;
+    }
+
+    return {
+      ...newState,
+      columnResizing: {
+        ...newState.columnResizing,
+        columnWidths: {
+          ...newState.columnResizing.columnWidths,
+          ...newColumnWidths,
+          ...newNextColumnWidths,
+        },
+      },
+    };
+  }
+
+  if (action.type === actions.columnDoneResizing) {
+    return {
+      ...newState,
+      columnResizing: {
+        ...newState.columnResizing,
+        startX: undefined,
+        isResizingColumn: undefined,
+      },
+    };
+  }
+  return newState;
+};
+
+const getColumnWidths = (
+  headerIdWidths: [string, number][],
+  deltaPercentage: number,
+) => {
+  const columnWidths: Record<string, number> = {};
+  headerIdWidths.forEach(([headerId, headerWidth]) => {
+    columnWidths[headerId] = Math.max(
+      headerWidth + headerWidth * deltaPercentage,
+      0,
+    );
+  });
+  return columnWidths;
+};
+
+const isNewColumnWidthsValid = <T extends Record<string, unknown>>(
+  columnWidths: Record<string, number>,
+  headers: ColumnInstance<T>[] | undefined,
+) => {
+  // Prevents from going outside the column bounds
+  if (Object.values(columnWidths).some((width) => width <= 1)) {
+    return false;
+  }
+
+  for (const [headerId, width] of Object.entries(columnWidths)) {
+    const header = headers?.find((h) => h.id === headerId);
+    if (!header) {
+      continue;
+    }
+
+    const minWidth = header.minWidth || 0;
+    const maxWidth = header.maxWidth || Infinity;
+    if (width < minWidth || width > maxWidth) {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+const useInstanceBeforeDimensions = <T extends Record<string, unknown>>(
+  instance: TableInstance<T>,
+) => {
+  const {
+    flatHeaders,
+    getHooks,
+    state: { columnResizing },
+  } = instance;
+
+  const getInstance = useGetLatest(instance);
+
+  flatHeaders.forEach((header, index) => {
+    const resizeWidth = columnResizing.columnWidths[header.id];
+    header.width = resizeWidth || header.width || header.originalWidth;
```

Comment:
react-table had `originalWidth` before `width`. Wonder what's the impact

Reply:
I think I did this on purpose, don't remember now. Because when resizing `width` and `originalWidth` will be different as `originalWidth` is the one set by the dev.